tests/numerics/fv/test_tpsa.py

```diff
@@ -21,1418 +36,1426 @@
-            "bound_mass_displacement": bound_mass_displacement,
-            "solid_mass_total_pressure": np.zeros((2, 2)),
-            # No contribution from cell center values to the boundary displacement
-            "bound_displacement_cell": np.zeros((4, 4)),
-            # The boundary displacement is the boundary condition
-            "bound_displacement_face": bound_displacement_face,
-            # Neither the rotation variable nor the solid pressure contribute to the
-            # boundary displacement for Dirichlet faces
-            "bound_displacement_rotation_cell": np.zeros((4, 2)),
-            "bound_displacement_solid_pressure_cell": np.zeros((4, 2)),
-        }
+    # The displacement should be close to the Neumann value.
+    assert np.allclose(x_rob_low, x_neu, rtol=1e-12)
 
-        self._compare_matrices(matrices, known_values)
 
-    def test_neumann_bcs(self):
-        """Set Neumann boundary conditions on all faces, check that the implementation
-        of the boundary conditions are correct.
-        """
-        _set_uniform_bc(self.g, self.data, "neu")
-        matrices = _discretize_get_matrices(self.g, self.data)
+@pytest.mark.parametrize("neu_bcs", [False, True])
+def test_3d_linear_displacement(neu_bcs: bool):
+    """Test of a 3d linear displacement problem, for which Tpsa should be exact.
 
-        # The values of the cell to face average for face for faces 0 and 6. Also the
-        # complements identified by suffix c_. On Neumann faces, only the cell value is
-        # used for the computation, thus it is assigned unit value (and the complement,
-        # with value 0, is ignored in the construction of the analytical expressions
-        # below, hence we don't define it).
-        c2f_avg_0 = 1
-        c2f_avg_6 = 1
+    The analytical solution is u = [z, 0, 0], which results in \sigma_{1,3} =
+    \sigma{3,1} = 1, with all other stress components equal to zero. The rotation will
+    be r = [0, -1, 0]. The solid pressure is zero.
 
-        # Boundary stress: The coefficients in bound_stress should be negative those of
-        # the stress matrix so that translation results in a stress-free configuration.
-        bound_stress = np.zeros((4, 14))
-        bound_stress[0, 0] = -1
-        bound_stress[1, 1] = -1
-        bound_stress[2, 12] = 1
-        bound_stress[3, 13] = 1
+    Boundary conditions can be set from this analytical solution, as either Dirichlet or
+    Neumann (controlled by the parameter neu_bcs). In both cases, the bottom boundary is
+    assigned homogenous Dirichlet conditions to ensure solvability.
 
-        bound_rotation_diffusion = np.zeros((2, 7))
-        bound_rotation_diffusion[0, 0] = -1
-        bound_rotation_diffusion[1, 6] = 1
+    """
+    # Notes for debugging: For Neumann BCs, at least two cells are needed in the x- or
+    # y-direction to fix the domain and ensure solvability.
+    g = pp.CartGrid([2, 1, 10], [1, 1, 1])
```

Comment:
Is run time negligible? Otherwise, consider 
```suggestion
    g = pp.CartGrid([2, 1, 4], [1, 1, 1])
```